### ISSUE TITLE
[bug] Broken layout

### ISSUE THREAD

**Issue by rafaell-lycan** [tadeuzagallo/GithubPulse#18]
@tadeuzagallo please check this.

I'll try to fix that later, but no sure.

![captura de tela 2015-01-14 as 15 48 52](https://cloud.githubusercontent.com/assets/1516450/5744171/62808722-9c05-11e4-810c-1e81a58c9201.png)

**Comment by tadeuzagallo** [comment]
Something similar have been reported on #11 but I haven't been able to reproduce yet...
It is a canvas element with fixed size, so it should happen. Really, it shouldn't...

Anyway, it's not a fix, but you can right click on the status bar icon, it'll reload the WebView and probably fix for now, so doesn't get stuck that way.

Did it just happen after your `commits today` went from `0` to `2`?

**Comment by rafaell-lycan** [comment]
@tadeuzagallo the last version 0.2.8 broken in native, but in Chrome it's ok.

What's happening? O.O

**Comment by tadeuzagallo** [comment]
What kind of broken?

**Comment by rafaell-lycan** [comment]
The same problem with the Graph.

**Comment by tadeuzagallo** [comment]
I think this might be happening because the canvas is the root view of the react component, I'll try wrapping it on another div and add a key to it and release on the next version

**Comment by tadeuzagallo** [comment]
@rafaell-lycan  I've tried to fix it altogether with the design improvements on the [experimental branch](https://github.com/tadeuzagallo/GithubPulse/tree/experimental/dist), if you'd please try it out and check if it still crashes for you that'd be much appreciated!

**Comment by rafaell-lycan** [comment]
@tadeuzagallo the font-family it's like Times? Yeah now it's ok but the font is strange.

**Comment by tadeuzagallo** [comment]
Damn it, I totally forgot to embed the font! haha
I'll fix it right away!

**Comment by rafaell-lycan** [comment]
@tadeuzagallo just wait to release this version. For now I think it's better make this fix in 0.2.9 with others stuffs, and in 0.3.0 maybe some others awesome features =)

**Comment by tadeuzagallo** [comment]
I just updated it, again on the [experimental branch](https://github.com/tadeuzagallo/GithubPulse/tree/experimental/dist), thanks for helping out!
The version is 0.3.0 because of the new layout, in order to be compliant with semver.org

**Comment by oscarmcm** [comment]
Some texts need to be adjusted @tadeuzagallo 
![image](https://cloud.githubusercontent.com/assets/2553459/5783328/7b3c91a6-9d87-11e4-94a6-f8f5a9331bbe.png)

**Comment by rafaell-lycan** [comment]
@oscarmcm I'm doing this fix. @tadeuzagallo you or someone take this?

**Comment by tadeuzagallo** [comment]
That had been fixed after v0.2.8, but the new layout came out and it never got released...
I'll move v0.3.0 from experimental to master and it should be ok!

**Comment by tadeuzagallo** [comment]
I'm closing it for now since v0.3.0 appears to have fixed it and is already available on master.
Ping me if you see it again and I'll reopen!

/cc @ljharb

**Comment by ljharb** [comment]
Thanks, manually updated (when's the Sparkle auto-update system coming?) and I'll let you know if it reoccurs

**Comment by rafaell-lycan** [comment]
Yeah, I want that too.

**Comment by ljharb** [comment]
This still happens to me on `v0.3.0` - although now it's a little different. 
![screen shot 2015-01-17 at 10 43 25 am](https://cloud.githubusercontent.com/assets/45469/5789544/b44ab424-9e35-11e4-96a8-5f6271385527.png)

**Comment by oscarmcm** [comment]
This still happens on v0.3.0 I have the same issue that @ljharb

**Comment by tadeuzagallo** [comment]
Really, how come that it never happened to me?!
I compile and recompile it hundreds of times everyday... I'm reopening and will try again to fix it.

**Comment by ljharb** [comment]
@tadeuzagallo are you on a non-retina mac? I'm on a Retina mac, and that almost certainly can affect your webview's CSS.

**Comment by oscarmcm** [comment]
@tadeuzagallo @ljharb yeah! I have a retina mac, maybe this is the problem, when react render the layout in the webview.

**Comment by tadeuzagallo** [comment]
@jlharb That should be it! Since @rafaell-lycan is also on o rMBP and I'm
not...
I'm looking into ways of simulating a retina display.
Thanks for the great catch!

**Comment by tadeuzagallo** [comment]
@ljharb was right! Thanks every body for helping out, I'll release it really soon.

TL;DR: The way I was using Chart.js was doubling the size of the canvas on every update 

Long explanation:
Chart.js checks if it's retina display on the constructor, and if it is, multiplies the size of the canvas per the screen pixel ratio. Every time the commits changes I created a new instance of Chart.js, so it was getting the bigger size, and would just keep increasing. The solution was dead easy though, just check if there was a previous instance and call `destroy` to restore it to its previous size before creating a new instance.

**Comment by rafaell-lycan** [comment]
@tadeuzagallo I like to use [amCharts](http://www.amcharts.com/), if you want, try it.
